header_test.go

```diff
@@ -11,6 +11,40 @@ import (
 	"testing"
 )
 
+func TestResponseHeaderMultiLineValue(t *testing.T) {
+	// This is a real WebLogic response
+	webLogicResponse := []byte("HTTP/1.1 200 OK\r\n" +
+		"Date: Mon, 04 Nov 2019 03:08:05 GMT\r\n" +
+		"Transfer-Encoding: chunked\r\n" +
+		"Content-Type: multipart/related; type=\"application/xop+xml\"; start=\"<soap.xml@xfire.codehaus.org>\"; start-info=\"text/xml\";\r\n" +
+		"	boundary=\"----=_Part_16_1220614102.1572836885552\"\r\n" +
+		"\r\n" +
+		"0528\r\n" +
+		"\r\n" +
+		"------=_Part_16_1220614102.1572836885552\r\n" +
+		"Content-Type: application/xop+xml; charset=UTF-8; type=\"text/xml\"\r\n" +
+		"Content-Transfer-Encoding: 8bit\r\n" +
+		"Content-ID: <soap.xml@xfire.codehaus.org>\r\n" +
+		"\r\n" +
+		"<soap:Envelope xmlns:soap=\"http://schemas.xmlsoap.org/soap/envelope/\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"><soap:Body><ns1:getMsgResponse xmlns:ns1=\"http://service.hundsun.com\"><ns1:out>&lt;?xml version=\"1.0\" encoding=\"utf-8\"?>&lt;ATS>&lt;PUB>&lt;TransSource>003&lt;/TransSource>&lt;TransCode>9144&lt;/TransCode>&lt;TransDate>20191017&lt;/TransDate>&lt;TransTime>132832&lt;/TransTime>&lt;TransSeq>201910171328320040&lt;/TransSeq>&lt;RtnCode>success&lt;/RtnCode>&lt;RtnMsg>处理成功&lt;/RtnMsg>&lt;/PUB>&lt;OUT>&lt;RD>&lt;RdSeq>191016218600407369&lt;/RdSeq>&lt;TransState>2&lt;/TransState>&lt;PayInfoCode>S0000&lt;/PayInfoCode>&lt;PayInfo>S0000-支付成功&lt;/PayInfo>&lt;CorpAct>1001280929023180000&lt;/CorpAct>&lt;CorpEntity>2000000&lt;/CorpEntity>&lt;CorpBank>102&lt;/CorpBank>&lt;OppAct>6216665500000178826&lt;/OppAct>&lt;PayMadeDate>20191101134056&lt;/PayMadeDate>&lt;Abstract>39101602215232&lt;/Abstract>&lt;ReqReserved1>&lt;/ReqReserved1>&lt;ReqReserved2>&lt;/ReqReserved2>&lt;/RD>&lt;/OUT>&lt;/ATS></ns1:out></ns1:getMsgResponse></soap:Body></soap:Envelope>\r\n" +
+		"002e\r\n" +
+		"\r\n" +
+		"------=_Part_16_1220614102.1572836885552--\r\n" +
+		"\r\n" +
+		"0000\r\n" +
+		"\r\n")
```

Comment:
Can you simplify this test a bit? It looks very complicated because the multipart body contains headers as well. But these are completely irrelevant to the test. Just have one header split on 2 lines and another header split on 3 lines to show both those cases work. And maybe a normal header before and after to show that those aren't affected (so all in 1 request).